### PR TITLE
Tuning performance testing

### PR DESCRIPTION
**Pull request by ligurio**
Depends on https://github.com/tarantool/tarantool/pull/10778
Depends on https://github.com/tarantool/tarantool/pull/10866

Tests in a branch: https://github.com/tarantool/tarantool/pull/10876

### CODE REVIEW

**Comment by ligurio** [comment]
Some checks made by `setup_env.sh` has failed:

```

Run sh ./perf/tools/setup_env.sh
Disable AMD SMT or Intel Hyperthreading  WARNING (hint: set 'nosmt' kernel parameter)
Isolate CPUs for benchmarking WARNING (hint: set 'isolcpus' kernel parameter)
Offload interrupts from the isolated CPUs CHECKED
Disable scheduling on single-task isolated CPUs CHECKED
Disable transparent huge pages CHECKED
Disable direct compaction of transparent huge pages CHECKED
Disable TurboBoost SKIPPED
Stabilize the frequency of CPU 0 SKIPPED
```

https://github.com/tarantool/tarantool/actions/runs/12054577379/job/33612934572?pr=10365

Should be fixed on hardware:

- https://github.com/tarantool/tarantool/wiki/Benchmarking#disable-amd-smt-or-intel-hyperthreading

**Comment by ligurio** [comment]
Disabled hyperthreading on hardware machine where benchmarks run:

  - edit `GRUB_CMDLINE_LINUX` in `/etc/default/grub`
  - run `update-grub`

After reboot:
```
~$ cat /sys/devices/system/cpu/smt/active
0
```

**Comment by ligurio** [comment]
### Offload interrupts from the isolated CPUs

- before: 
```
$ cat /proc/irq/default_smp_affinity
ffffffff
```
- added ` irqaffinity=0-15` to `GRUB_CMDLINE_LINUX` in `/etc/default/grub`
- run `update-grub`
- after reboot:
```
$ cat /proc/irq/default_smp_affinity
0000ffff
```

### Isolated CPUs for benchmarking

- before:
```
$ cat /sys/devices/system/cpu/isolated

$
``` 
- add `isolcpus=16-31 ` to `GRUB_CMDLINE_LINUX` in `/etc/default/grub`
- run `update-grub`
- after reboot:
```
$ cat /sys/devices/system/cpu/isolated
16-31
$
```

**Comment by ligurio** [comment]
Fixed running `setup_env` on regular runners:

```diff
--- a/.github/workflows/perf_micro.yml
+++ b/.github/workflows/perf_micro.yml
@@ -85,6 +85,11 @@ jobs:
           submodule: ${{ inputs.submodule }}
           revision: ${{ inputs.revision }}
       - name: setup environment
+        # Run on push to the 'master' and release branches or on
+        # non-fork pull requests. The script will fail on regular
+        # runners.
+        if: github.event_name != 'pull_request' ||
+          github.event.pull_request.head.repo.full_name == github.repository
         run: sh ./perf/tools/setup_env.sh
       - name: test
         run: make -f .test.mk test-perf
```

File: .github/workflows/perf_micro.yml [hunk 1]
```diff
@@ -97,7 +97,7 @@ jobs:
           # sudo taskset 0xef chrt 50.
           # But this makes the process use non-standard, real-time
           # round-robin scheduling mechanism.
-          BENCH_CMD: "taskset 0xfe chrt 50"
+          BENCH_CMD: "taskset 0x01fffffe chrt 50"
```
Comment (Buristan) [inline .github/workflows/perf_micro.yml:105]: Minor: I would add the comment here about the specific mask ("Hardware machine for benchmarking has a group of isolated CPUs 1-24.") from the commit message.

**Review by Buristan — APPROVED** [review]
Hi, Sergey!
Thanks for the patch!
LGTM with a minor ignorable comment below.

File: .github/workflows/perf_micro.yml (hunk 1 above)
Comment (ligurio) [reply .github/workflows/perf_micro.yml:105]: Fixed:

```diff
--- a/.github/workflows/perf_micro.yml
+++ b/.github/workflows/perf_micro.yml
@@ -102,6 +102,8 @@ jobs:
           # sudo taskset 0xef chrt 50.
           # But this makes the process use non-standard, real-time
           # round-robin scheduling mechanism.
+          # Hardware machine for benchmarking has a group of
+          # isolated CPUs 1-24.
           BENCH_CMD: "taskset 0x01fffffe chrt 50"
       - name: Aggregate benchmark results
         run: make -f .test.mk test-perf-aggregate
```

**Review by opomuc — APPROVED** [review]

**Pull request merged**